lightly_studio/tests/api/routes/api/test_dataset.py

```diff
@@ -92,6 +92,51 @@ def test_read_root_dataset__multiple_root_datasets(
     assert response.status_code != HTTP_STATUS_OK
 
 
+def test_read_dataset_hierarchy(test_client: TestClient, db_session: Session) -> None:
+    """Test dataset hierarchy retrieval.
+
+    - A (root)
+      - B
+        - C
+      - D
+    """
+    client = test_client
+    dataset_a_id = create_dataset(session=db_session, dataset_name="example_dataset").dataset_id
+    dataset_b_id = create_dataset(
+        session=db_session, dataset_name="child_B", parent_dataset_id=dataset_a_id
+    ).dataset_id
+    dataset_c_id = create_dataset(
+        session=db_session, dataset_name="child_C", parent_dataset_id=dataset_b_id
+    ).dataset_id
+    dataset_d_id = create_dataset(
+        session=db_session, dataset_name="child_D", parent_dataset_id=dataset_a_id
+    ).dataset_id
```

Comment:
Side note: I named the test datasets `ds_a`, etc. for a bit more space efficiency and hopefully readability :) It is fine as is here.